File: src/main/java/com/hoi4utils/clausewitz/data/country/Character.java
package com.hoi4utils.clausewitz.data.country;

import com.hoi4utils.clausewitz.localization.*;
import scala.collection.Iterable;
import scala.jdk.javaapi.CollectionConverters;

import java.util.List;
import java.util.Map;


public class Character implements Localizable {
    String id;

    @Override
    public scala.collection.mutable.Map<Property, String> getLocalizableProperties() {
        return CollectionConverters.asScala(Map.of(Property.NAME, id));
    }

    @Override
    public Iterable<? extends Localizable> getLocalizableGroup() {
        return CollectionConverters.asScala(List.of(this)); // todo improve
    }
}

File: src/main/scala/com/hoi4utils/hoi4/country/Character.scala
package com.hoi4utils.hoi4.country

import com.hoi4utils.localization.{Localizable, Property}

import scala.collection.mutable
import scala.collection.mutable.Map
import scala.jdk.javaapi.CollectionConverters

class Character extends Localizable {
  private val id: String = null

  override def getLocalizableProperties: mutable.Map[Property, String] = mutable.Map(Property.NAME -> id)

  override def getLocalizableGroup: Iterable[? <: Localizable] = List(this) // todo improve
}

